test/object-store/c_api/c_api.cpp

```diff
@@ -4090,6 +4090,7 @@ TEST_CASE("C API - client reset", "[c_api][client-reset]") {
         return reset_utils::make_baas_client_reset(config_local, config_remote, test_app_session);
     };
 
+    local_config.sync_config = std::make_shared<realm_sync_config_t>(*local_config.sync_config);
```

Comment:
Casting something allocated as SyncConfig to a realm_sync_config_t is invalid. The test happened to work because it didn't call anything which would use the uninitialized vtable ptr.